app/components/IconButton.js

```diff
@@ -15,26 +15,26 @@ class IconButton extends Component<Props> {
 
     return (
       <View style={styles.view}>
-        {this._selectIcon()}
+        <Icon
+          name={this._selectIcon(name)}
+          style={[styles.icon, { color: tint }]}
+        />
         <Text style={[styles.text, { color: tint }]}>{name}</Text>
       </View>
     );
   }
 
-  _selectIcon() {
-    const { name, tint } = this.props;
-
-    if (name == "Goals") {
-      return <Icon name="md-filing" style={[styles.icon, { color: tint }]} />;
-    }
-
-    if (name == "Progress") {
-      return (
-        <Icon name="md-speedometer" style={[styles.icon, { color: tint }]} />
-      );
+  _selectIcon(name: string) {
```

Comment:
This is kind of personal preference, but if you're interested in making this _actually_ private (as opposed to pseudo-private, indicated by the leading `_`) then you could pull this out to a function defined (but not exported) in this file. It's completely pure (i.e. no side-effects, doesn't care about any state) so it doesn't _need_ to be a method on the class.